### PR TITLE
AG try debugging w extend

### PR DESCRIPTION
**Pull request by adonomay**
needs testing of haploid and homoz-only -w_extend with multiple dogs.

Edit: have tested this in bulk on dobermans and ref panel v7, looks good (eCOI is consistent with measured COI)

### CODE REVIEW

**Review by svohr — COMMENTED** [review]
Can you highlight what you changed in the logic of the program besides debug prints/variable/names/style? It's difficult to see how the behavior has changed.

File: GERMLINE_0001.cpp [hunk 1]
```diff
@@ -90,7 +95,7 @@ int main(int argc, char* argv[])
 
 	GERMLINE germline;
     germline.mine( params );
-    return 1;
+    return 0;
```
Comment (ajsams) [inline GERMLINE_0001.cpp:98]: There's a spot in the pipeline where I remember we were handling this bug, so we'll need to fix that.

File: Match.cpp [hunk 2]
```diff
@@ -70,76 +70,154 @@ bool Match::approxEqual()
...
 		for ( int a = 0 ; a < 2 ; a++ ) {
 			for ( int b = 0 ; b < 2 ; b++ ) { 
-				err = false;
-				for ( cur_marker = MARKER_SET_SIZE - 1 ; cur_marker >= 0 && ! err; cur_marker-- )
+				found_mismatch = false;
+				for ( cur_marker = MARKER_SET_SIZE - 1; cur_marker >= 0 && !found_mismatch; cur_marker-- )
 				{
 					if ( node[0]->getChromosome( a )->getMarkerSet()->getMarkerBits()[cur_marker] != node[1]->getChromosome( b )->getMarkerSet()->getMarkerBits()[cur_marker] )
-						err = true;
+						found_mismatch = true; // err may actually mean end-of-homozygous-tract?
```
Comment (ajsams) [inline Match.cpp:116]: It looks that way

File: readme.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+### Change log
+
+#### Version 1.5.2
+
+* replicates stream-reading fix that David Riccardi made in 2016 (but didn't make into version control)
+* `germline -haploid -w_extend -bits 41` (or any bits value > 1) now produces correct match tracts
+* exit codes follow shell-script convention: 1 for error, 0 for success
+* `germline -version` now reports the version number
+* added tests
```
Comment (ajsams) [inline readme.md:81]: Nice

**Review by ajsams — APPROVED** [review]
Looks good to me, admittedly, I'm not fluent in C++ so don't understand all of this.

**Comment by adonomay** [comment]
@svohr the one change was passing the marker_set_number parameter to the getMarkerSet function call in lines 85 and 86 of Match.cpp

**Pull request merged**